File: 110. Balanced Binary Tree/recursion.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def maxDepth(self,root):
        if root:
            return max(self.maxDepth(root.left)+1,self.maxDepth(root.right)+1)
        return 0;
    
    def isBalanced(self, root):
        """
        :type root: TreeNode
        :rtype: bool
        """
        if root:
            if abs(self.maxDepth(root.left)-self.maxDepth(root.right))<=1:
                return self.isBalanced(root.left) and self.isBalanced(root.right)
            else:
                return False
        return True

File: 110. Balanced Binary Tree/recursion.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     struct TreeNode *left;
 *     struct TreeNode *right;
 * };
 */

int max(int a, int b){
    if (a>b) return a;
    else return b;
}
int maxDepth(struct TreeNode* p){
    if (p) return max(1+maxDepth(p->right),1+maxDepth(p->left));
    return 0;
}
int abs1(int a, int b){
    if (a>b) return a-b;
    else return b-a;
}

bool isBalanced(struct TreeNode* root) {
    if (root){
        if (abs1(maxDepth(root->left),(maxDepth(root->right)))<=1) return isBalanced(root->left)&&isBalanced(root->right);
        else return false;
    }
    return true;
    
}
